pkgs/games/openloco/default.nix

```
@@ -0,0 +1,32 @@
+{ fetchFromGitHub, lib, stdenv, cmake, libpng, libzip, openal, SDL2, yaml-cpp, span-lite, gtest, pkg-config }:
+
+stdenv.mkDerivation rec {
+  pname = "openloco";
+  version = "23.02";
+
+  src = fetchFromGitHub {
+    owner = "OpenLoco";
+    repo = "OpenLoco";
+    rev = "v${version}";
+    hash = "sha256-35g7tnKez4tnTdZzavfU+X8f3btFG6EbLkU+cqL6Qek=";
+  };
+
+  # the upstream build process determines the version tag from git, since `src` is not a git checkout we just set it manually
+  postPatch = ''
+    substituteInPlace src/OpenLoco/src/Version.cpp --replace "#define NAME \"OpenLoco\"" '#define NAME "OpenLoco"
+    #define OPENLOCO_VERSION_TAG "${version}"'
+  '';
+
+  NIX_CFLAGS_COMPILE = "-Wno-error=null-dereference";
+
+  buildInputs = [pkg-config libpng libzip openal SDL2 yaml-cpp span-lite gtest];
+  nativeBuildInputs = [cmake];
```

```suggestion
  nativeBuildInputs = [
    cmake
    pkg-config
  ];

  buildInputs = [
    SDL2
    libpng
    libzip
    openal
    yaml-cpp
    span-lite 
    gtest
  ];
```

P.S.: is gtest truly needed?

---

Found the flag to disable building tests